File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/ResponseTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class ResponseTyped {
    public ResultTyped result;

    public ResponseTyped(ResultTyped result) {
        this.result = result;
    }

    public ResponseTyped(ResponseString s) {
        this.result = new ResultTyped(s.result);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ResponseTyped)) return false;
        ResponseTyped that = (ResponseTyped) o;
        return Objects.equals(result, that.result);
    }

    @Override
    public int hashCode() {
        return Objects.hash(result);
    }

    public ResponseString toResponseString() {
        return new ResponseString(result.toResultString());
    }

    public static List<ResponseTyped> fromStringList(List<ResponseString> list) {
        List<ResponseTyped> result = new ArrayList<>();
        for (ResponseString s : list) result.add(new ResponseTyped(s));
        return result;
    }

    public static List<ResponseString> toStringList(List<ResponseTyped> list) {
        List<ResponseString> result = new ArrayList<>();
        for (ResponseTyped t : list) result.add(t.toResponseString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("result", result == null ? null : result.toJsonValue());
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static ResponseTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new ResponseTyped(
            ResultTyped.fromJsonValue(Json.asObject(Json.require(m, "result"), "result")));
    }

    public static ResponseTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<ResponseTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (ResponseTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<ResponseTyped> fromJSONList(String json) {
        List<ResponseTyped> result = new ArrayList<ResponseTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "ResponseTyped")));
        return result;
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/common/ResponseTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class ResponseTyped
    {
        public ResultTyped result;

        public ResponseTyped(ResultTyped result)
        {
            this.result = result;
        }

        public ResponseString ToResponseString()
        {
            return new ResponseString(
                this.result.ToResultString()
            );
        }

        public static List<ResponseString> ToStringList(List<ResponseTyped> list)
        {
            var result = new List<ResponseString>();
            foreach (var t in list) result.Add(t.ToResponseString());
            return result;
        }

        public static List<ResponseTyped> FromStringList(List<ResponseString> list)
        {
            var result = new List<ResponseTyped>();
            foreach (var s in list) result.Add(s.ToResponseTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WritePropertyName("result");
            this.result.WriteJson(w);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static ResponseTyped FromJsonElement(JsonElement m)
        {
            return new ResponseTyped(
                ResultTyped.FromJsonElement(Json.Require(m, "result"))
            );
        }

        public static ResponseTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<ResponseTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<ResponseTyped> FromJSONList(string json)
        {
            var result = new List<ResponseTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "ResponseTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"result={result}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not ResponseTyped other) return false;
            return object.Equals(this.result, other.result);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.result);
            return h.ToHashCode();
        }
    }
}
